cmd/end2endtest/helpers.go

```diff
@@ -438,71 +455,112 @@ func (t *e2eElection) overwriteVote(choices []int, indexAcct int, waitType strin
 }
 
 // ballotVotes from a default list of 10 vote values that exceed the max value, max total cost and is not unique will
-func ballotVotes(b ballotData, nvotes int) ([][]int, [][]*types.BigInt) {
+func ballotVotes(vop vapi.TallyMode, nvotes int, ballotType string, expectUniqueV bool) ([][]int, [][]*types.BigInt) {
 	votes := make([][]int, 0, nvotes)
-	var resultsField1, resultsField2 []*types.BigInt
-
-	// initial 10 votes to be sent by the ballot test
-	var v = [][]int{
-		{0, 0}, {0, 7}, {5, 7}, {2, 0}, {2, 3},
-		{1, 6}, {0, 8}, {6, 5}, {2, 7}, {6, 4},
+	var resultsFields [][]*types.BigInt
+	var v [][]int
+
+	switch ballotType {
+	// fill initial 10 votes to be sent by the ballot test
+	case "ranked":
+		v = [][]int{
+			{0, 5, 0, 2}, {3, 1, 0, 2}, {3, 2, 0, 1}, {1, 0, 3, 2},
+			{2, 3, 1, 1}, {4, 1, 1, 0}, {0, 0, 2, 1}, {0, 2, 1, 3},
+			{1, 1, 1, 4}, {0, 3, 2, 1},
+		}
+		// default results on field 1,2,3,4 for 10 votes
+		resultsFields = append(resultsFields, votesToBigInt(20, 10, 0, 20),
+			votesToBigInt(10, 10, 20, 10), votesToBigInt(20, 10, 10, 10),
+			votesToBigInt(0, 20, 20, 10))
+
+	case "quadratic":
+		v = [][]int{
+			{2, 4, 2, 0}, {3, 3, 2, 2}, {2, 3, 0, 1}, {0, 2, 3, 3},
+			{1, 2, 1, 2}, {5, 0, 1, 0}, {0, 2, 2, 2}, {2, 1, 1, 2},
+			{1, 3, 0, 1}, {1, 3, 1, 1},
+		}
+		resultsFields = append(resultsFields, votesToBigInt(10, 30, 10, 0),
+			votesToBigInt(0, 10, 20, 20), votesToBigInt(10, 30, 10, 0),
+			votesToBigInt(0, 20, 30, 0))
+
+	case "range":
+		v = [][]int{
+			{5, 0, 4, 2}, {3, 1, 2, 4}, {6, 0, 1, 3}, {0, 3, 2, 1},
+			{2, 1, 3, 0}, {3, 2, 1, 3}, {1, 3, 2, 0}, {2, 1, 1, 2},
+			{1, 2, 0, 3}, {4, 3, 1, 1},
+		}
+		resultsFields = append(resultsFields, votesToBigInt(10, 20, 10, 0),
+			votesToBigInt(0, 10, 10, 20), votesToBigInt(10, 0, 20, 10),
+			votesToBigInt(20, 10, 0, 10))
+
+	case "approval":
+		v = [][]int{
+			{1, 0, 0, 0}, {2, 1, 0, 1}, {0, 1, 2, 3}, {0, 0, 0, 1},
+			{0, 0, 1, 1}, {2, 2, 1, 0}, {1, 0, 1, 1}, {0, 0, 0, 0},
+			{1, 1, 0, 3}, {1, 1, 1, 1},
+		}
+		resultsFields = append(resultsFields, votesToBigInt(30, 30, 0, 0),
+			votesToBigInt(50, 10, 0, 0), votesToBigInt(30, 30, 0, 0),
+			votesToBigInt(20, 40, 0, 0))
 	}
```

Comment:
i have the feeling this block would fit better in ballot.go, 
i.e. make Run() evaluate `b.ballotType` and pass `v` and `resultsFields` to `ballotVotes`

Reply:
uhm, maybe adding two more parameters to `ballotVotes` function that currently has 4 parameters could complicate the things instead of simplify :thinking: